tracer/src/Datadog.Tracer.Native/clr_helpers.h

```diff
@@ -671,6 +671,43 @@ HRESULT IsTypeByRefLike(ICorProfilerInfo4* corProfilerInfo4, const ModuleMetadat
 
 HRESULT IsTypeTokenByRefLike(ICorProfilerInfo4* corProfilerInfo4, const ModuleMetadataBase& module_metadata, mdToken typeDefOrRefOrSpecToken,
                              bool& isTypeIsByRefLike);
+
+template<class MetaInterface>
+class COMPtrHolder {
+public:
+    COMPtrHolder() {
+        m_ptr = NULL;
+    }
+
+    COMPtrHolder(MetaInterface *ptr) {
+        if (ptr != NULL) {
+            ptr->AddRef();
```

Comment:
I don't think you're supposed to call AddRef here, isn't the interface that you receive already AddRef'd?

Reply:
yes you're right:

https://github.com/dotnet/runtime/blob/v8.0.11/src/coreclr/vm/proftoeeinterfaceimpl.cpp#L10498

then

https://github.com/dotnet/runtime/blob/1b8518435bde23806d850ab399828f8f594e95a0/src/coreclr/vm/profilingenumerators.h#L460

then

https://github.com/dotnet/runtime/blob/1b8518435bde23806d850ab399828f8f594e95a0/src/coreclr/vm/profilingenumerators.h#L99

Reply:
fixed here: [876fa79](https://github.com/DataDog/dd-trace-dotnet/pull/6334/commits/876fa79b10a8b038bc05d1717acdd077b07c4459)